### PR TITLE
[#852] Zoom tab displays commits not within filter time rage

### LINKED ISSUE #852: Zoom tab displays commits not within filter time
Current: 
Instead of zooming in terms of the position of the dotted lines, the current zoom feature gets the nearest since and until dates from the dotted lines, and zooms into commits that are within the duration.

Issue: 
Ramps that are not included in the dotted lines are also included in the zoom tab.

Screen Shot 2019-08-01 at 11 48 03 PM

### PR DESCRIPTION
**Pull request by fzdy1914**
Fixes #852

When we try to open the zoom tab for the same user with the same 
date range for the second time, commits displayed the second time 
contain commit that does not fall within the filter time range.

This is mainly due to we edit the `user` object of `info` property 
directly. However, `info` is the property of the child component so
 we are not supposed to change it directly. The situation mentioned 
above occurs because this property is overwritten again by the 
parent, which related to the way we re-render the sidebar.

Let's use another computed value to represent the user with filtered
 commits so we do not need to modify the property directly. Minor 
changes are done together to improve code quality.

### CODE REVIEW

File: frontend/src/tabs/zoom.pug [hunk 1]
```diff
@@ -1,31 +1,32 @@
...
-      span(v-else) {{ info.user.repoName }}
+      span(v-if="info.filterGroupSelection === 'groupByAuthors'") {{ filteredUser.displayName }} ({{ filteredUser.name }})
+      span(v-else) {{ filteredUser.repoName }}
     .zoom__title--chart.medium-font(v-if="!info.isMergeGroup")
       span &#8627; &nbsp;
-      span(v-if="info.filterGroupSelection === 'groupByAuthors'") {{ info.user.repoName }}
-      span(v-else) {{ info.user.displayName }} ({{ info.user.name }})
+      span(v-if="info.filterGroupSelection === 'groupByAuthors'") {{ filteredUser.repoName }}
+      span(v-else) {{ filteredUser.displayName }} ({{ filteredUser.name }})
+      span(v-else) {{ filteredUser.displayName }} ({{ filteredUser.name }})
```
Comment (jamessspanggg) [inline frontend/src/tabs/zoom.pug]: Is this repeated?

Comment (fzdy1914) [reply frontend/src/tabs/zoom.pug]: My bad.

**Comment by fzdy1914** [comment]
Fixed the issue mentioned in #948

**Comment by Tejas2805** [comment]
One nit. 

The `info.user` here should be `filteredUser`

Screen Shot 2020-01-02 at 4 44 05 PM

**Comment by Tejas2805** [comment]
Also, can you update the commit message?

**Comment by fzdy1914** [comment]
> Also, can you update the commit message?

I don't feel that the commit message needs to be updated as the changes are quite small. It is more like improving our code quality.

File: frontend/src/static/js/v_zoom.js [hunk 2]
```diff
@@ -5,41 +5,44 @@ window.vZoom = {
...
     },
-
-    getCommitMessageBodyCount() {
+    totalCommitMessageBodyCount() {
       let nonEmptyCommitMessageCount = 0;
-      this.info.user.commits.forEach((commit) => {
+      this.filteredUser.commits.forEach((commit) => {
         commit.commitResults.forEach((commitResult) => {
-          if (commitResult.messageBody !== '') {
+          if (commitResult.messageBody !== '' && commitResult.insertions > 0) {
```
Comment (fzdy1914) [inline frontend/src/static/js/v_zoom.js]: This line is added as we only display commits with insertions > 0

**Comment by Tejas2805** [comment]
@eugenepeh @yong24s For your final review.

**Pull request merged**